### PR TITLE
Add support for scaling based on RouteGroup resource

### PR DESCRIPTION
**Pull request by mikkeloscar**
Extends the skipper collector to also support referencing a [`RouteGroup`](https://opensource.zalando.com/skipper/kubernetes/routegroups/) resource for scaling based on RPS.

The weights are based on the `defaultBackends` defined for the routegroup. It does not try to handle specific weighting per path or so which is possible in RouteGroup but is not the common use case. This would be compatible with the ingress RPS scaling we have on StackSets already.

### CODE REVIEW

File: pkg/collector/skipper_collector.go [hunk 1]
```diff
@@ -128,26 +134,65 @@ func getWeights(ingressAnnotations map[string]string, backendAnnotations []strin
...
-	if err != nil {
-		return nil, err
+	for _, backend := range backends {
+		if backend.BackendName == backendName {
+			return float64(backend.Weight) / 100.0, nil
+		}
 	}
-	config := c.config
 
+	if backendName != "" {
```
Comment (jonathanbeber) [inline pkg/collector/skipper_collector.go:148]: Is it possible for `backend.BackendName` to be empty? I don't think so.
I'd suggest moving this check to the top and invert (i.e. `if backendName == ""`) and returning earlier, don't see why to range over all backends to figure out an empty backend name was provided.

File: pkg/collector/skipper_collector.go [hunk 2]
```diff
@@ -128,26 +134,65 @@ func getWeights(ingressAnnotations map[string]string, backendAnnotations []strin
...
-		escapedHostnames = append(escapedHostnames, regexp.QuoteMeta(strings.Replace(rule.Host, ".", "_", -1)))
+	var backendWeight float64
+	switch c.objectReference.Kind {
+	case "Ingress":
+		ingress, err := c.client.NetworkingV1beta1().Ingresses(c.objectReference.Namespace).Get(ctx, c.objectReference.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		backendWeight, err = getWeights(ingress.Annotations, c.backendAnnotations, c.backend)
```
Comment (jonathanbeber) [inline pkg/collector/skipper_collector.go:166]: :nail_care: nit: rename to 
```suggestion
		backendWeight, err = getIngressWeight(ingress.Annotations, c.backendAnnotations, c.backend)
```

File: pkg/collector/skipper_collector.go (hunk 1 above)
Comment (mikkeloscar) [reply pkg/collector/skipper_collector.go]: Right, changed!

File: pkg/collector/skipper_collector.go (hunk 2 above)
Comment (mikkeloscar) [reply pkg/collector/skipper_collector.go]: changed!

**Comment by jonathanbeber** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**